pkg/kube/kroler_test.go

```diff
@@ -0,0 +1,29 @@
+package kube
+
+import (
+	"context"
+	"testing"
+
+	"github.com/ash2k/iam4kube/pkg/util/logz"
+	"github.com/stretchr/testify/assert"
+	"k8s.io/client-go/tools/cache"
+)
+
+func mockIndexFunc(obj interface{}) (string, error) {
+	return "foo", nil
+}
+
+func TestRoleForIpNotFound(t *testing.T) {
+	store := cache.NewIndexer(mockIndexFunc, cache.Indexers{podByIpIndex: podByIpIndexFunc})
+	store2 := cache.NewIndexer(mockIndexFunc, cache.Indexers{})
+
+	kroler := Kroler{
+		logger:    logz.DevelopmentLogger(),
+		podIdx:    store,
+		svcAccIdx: store2,
+	}
+
+	role, err := kroler.RoleForIp(context.Background(), "127.0.0.1")
+	assert.Nil(t, role)
```

Comment:
By convention if error is not nil other values should not be relied upon. So there is no contract that role will be nil.